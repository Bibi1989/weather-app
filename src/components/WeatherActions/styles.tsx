import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 0px;
  width: 100%;

  @media (max-width: 768px) {
    flex-direction: column;

    &.btn {
      margin-top: 10px;
    }
  }
`;

export const Wrapper = styled.div``;
