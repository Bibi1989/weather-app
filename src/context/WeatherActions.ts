export const GET_WEATHERS = "GET_WEATHERS";
export const LOADING = "LOADING";

export const getWeathersAction = <T>(payload: T) => {
  return {
    type: GET_WEATHERS,
    payload,
  };
};

export const loadingAction = (payload: boolean) => {
  return {
    type: LOADING,
    payload,
  };
};
