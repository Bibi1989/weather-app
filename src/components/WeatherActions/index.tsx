import React from "react";

import CustomBtn from "components/UI/CustomBtn";
import RadioGroup from "components/UI/RadioGroup";
import { WeatherActionProps } from "typescript/react-props.types";
import { Container, Wrapper, Label } from "./styles";
import { CityTag } from "pages/styles";

const WeatherActions: React.FC<WeatherActionProps> = ({
  refreshData,
  onChange,
  units,
  city,
  currentLocation,
}) => {
  return (
    <Wrapper>
      <Container>
        <div></div>
        <RadioGroup onChange={onChange} units={units} />
        <CustomBtn label="Refresh" onClick={refreshData} testId="refreshBtn" />
      </Container>
      <Container>
        <div>
          <CustomBtn
            label="Select Current Location"
            onClick={currentLocation}
            testId="currentLocationBtn"
          />
          <Label>Note: Enable location in your browser</Label>
        </div>
        <CityTag>Location: {city?.name}</CityTag>
      </Container>
    </Wrapper>
  );
};

export default WeatherActions;
