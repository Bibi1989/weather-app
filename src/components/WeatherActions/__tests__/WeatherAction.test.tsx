import React from "react";
import { render, fireEvent } from "@testing-library/react";

import WeatherActions from "..";
import { WeatherActionProps } from "typescript/react-props.types";

const renderWeatherActions = (props: Partial<WeatherActionProps> = {}) => {
  const defaultProps: WeatherActionProps = {
    refreshData() {
      return;
    },
    onChange() {
      return;
    },
    units: "metric",
  };

  return render(<WeatherActions {...defaultProps} {...props} />);
};

describe("WeatherAction", () => {
  it("Refresh button snapshot", () => {
    const { container } = renderWeatherActions();

    expect(container).toMatchSnapshot();
  });
  it("Refresh button should be in the document", () => {
    const { queryByTestId } = renderWeatherActions();

    const refreshBtn = queryByTestId("refreshBtn");

    expect(refreshBtn).toBeInTheDocument();
  });
  it("refreshData should be called", () => {
    const refreshData = jest.fn();
    const { getByTestId } = renderWeatherActions({ refreshData });

    const refreshBtn = getByTestId("refreshBtn");

    fireEvent.click(refreshBtn);

    expect(refreshData).toHaveBeenCalledTimes(1);
  });
});
