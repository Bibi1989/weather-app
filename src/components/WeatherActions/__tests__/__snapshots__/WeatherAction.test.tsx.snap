// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherAction Refresh button snapshot 1`] = `
<div>
  <div
    class="sc-jRQBWg bBjQle"
  >
    <div />
    <div
      class="sc-bdvvtL cwkCTu"
    >
      <div
        class="sc-gsDKAQ kIvtJh input_group"
      >
        <input
          checked=""
          id="c"
          name="degree"
          type="radio"
          value="metric"
        />
        <label
          for="c"
        >
          Celcius
        </label>
      </div>
      <div
        class="sc-gsDKAQ kIvtJh"
      >
        <input
          id="f"
          name="degree"
          type="radio"
          value="imperial"
        />
        <label
          for="f"
        >
          Fahrenheit
        </label>
      </div>
    </div>
    <div
      class="sc-eCImPb uljmJ btn"
      data-testid="refreshBtn"
    >
      Refresh
    </div>
  </div>
</div>
`;
